### PR TITLE
SkypeEventLoop patches

### PR DESCRIPTION
**Pull request by codeofdusk**
This pull request makes SkypeEventLoop more flexible for programs that deal with more than just Skype events (they can add their own logic to the loop). It also fixes an AttributeError I was experiencing.

### CODE REVIEW

File: skpy/main.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ def cycle(self):
         """
         try:
             events = self.getEvents()
-        except requests.ConnectionError:
+        except (requests.ConnectionError,AttributeError) as e:
```
Comment (OllieTerrance) [inline skpy/main.py]: Not a good idea to catch that, if you're seeing one then it's probably a bug in the event parsing.  Can you reproduce the error and drop a stack trace in the comments?

**Comment by codeofdusk** [comment]
Traceback (most recent call last):
  File "looptest.py", line 11, in <module>
    s.loop()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/SkPy-0.0.0-py3.5.egg/skpy/main.py", line 193, in loop
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/SkPy-0.0.0-py3.5.egg/skpy/conn.py", line 77, in wrapper
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/SkPy-0.0.0-py3.5.egg/skpy/conn.py", line 77, in wrapper
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/SkPy-0.0.0-py3.5.egg/skpy/main.py", line 118, in getEvents
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/SkPy-0.0.0-py3.5.egg/skpy/event.py", line 52, in fromRaw
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/SkPy-0.0.0-py3.5.egg/skpy/event.py", line 88, in rawToFields
AttributeError: 'SkypeEnum' object has no attribute 'Away'

**Comment by OllieTerrance** [comment]
Okay, so you're seeing presence events where the user has an "Away" status.  Which is odd, as I've only ever seen "Idle", even with my test account now set as away...

Either they're rolling out a rename for that status, or there's a way to set either status (maybe old clients).  In any case, I'll add it as a known status, which should stop that error from cropping up.

**Comment by codeofdusk** [comment]
Thanks. Will you merge my first commit? Would be useful in some things I’m working on and could be valuable to others as well.

Bill

**Comment by OllieTerrance** [comment]
👍

**Pull request closed without merging**